analyses/independent-samples-release/util/independent-rna-samples.R

```diff
@@ -0,0 +1,118 @@
+# independent-rna-samples.R
+
+#' Generate a vector of unique rna samples
+#' 
+#' The samples from this function will be unique with respect to participants
+#' i.e. only no two samples will come from the same participant. The input list 
+#' should be pre-filtered by `composition` and `sample_type`.  
+#' 
+#' 
+#' @param independent_dna_sample_df A data frame of samples, with columns 
+#' corresponding to those in `independent-specimens.wgswxspanel.primary.tsv` 
+#' or `independent-specimens.wgswxspanel.primary-plus.tsv` depending on what 
+#' set of samples you need to include
+#' @param histology_df A data frame of samples, with columns corresponding 
+#' to those `histologies.tsv`
+#' @param match_type Designates which type matching needs to be done. Options
+#' are "independent_dna" to include only rna samples that match the 
+#' independent-specimens.wgswxspanel sample set, 
+#' "independent_dna_plus_only_rna" to include samples that macth the dna sample 
+#' set plus include samples where only rna samples exists
+#' @param tumor_description_rna_only Tumor descriptors to select samples where
+#' only RNA samples are available and will have no matching id in independent_dna_sample_df
+#' Opetions are "primary" to select only primary/initial tumors As of v18, primary tumors are defined as those designated "Initial CNS Tumor" .
+#' "primary_plus" if you would like to select other non-initial tumor RNA-Seq sample if no 
+#' initial tumor RNA-Seq sample exists
+#' or "Diagnosis" in the `tumor_descriptor` field.
+#' @param seed An optional random number seed. 
+#' 
+#' @return a data frame of Participant and Specimen IDs, each present only once.
+independent_rna_samples <- function(independent_dna_sample_df, 
+                                histology_df,
+                                match_type = c("independent_dna", "independent_dna_plus_only_rna"),
+                                tumor_description_rna_only = c("primary","primary_plus"),
+                                seed){
+  match_type <- match.arg(match_type)
+  tumor_description_rna_only <- match.arg(tumor_description_rna_only)
+  if(!missing(seed)){set.seed(seed)}
+  primary_descs <- c("Initial CNS Tumor", "Diagnosis", "Primary Tumor")
+  
+  
```

Comment:
```suggestion
  histology_df <- histology_df %>%
    dplyr::mutate(match_id = paste(Kids_First_Participant_ID, sample_id, sep = "_sample_id_"))
```

As `match_id` column is not added in `histology_df`, the column will have `NA`s in `match_type == "independent_dna_plus_only_rna"` and `match_type == "independent_dna_plus_only_rna" & tumor_description_rna_only == "primary_plus"` conditions.

Related to `match_id` comment in independent DNA function.

Reply:
If `match_id` is added here  to `histology_df` no need to add it subsequent set step as @logstar has illustrated